source/State.hpp

```diff
@@ -442,4 +418,74 @@ class DMoorDynStateDt
 	void Mix(const DMoorDynStateDt& visitor, const real& f);
 };
 
+/**
+ * @brief Do the computation for a row of a Butcher Tableau for an explicit
+ * integrator
+ *
+ * This function unwraps the computation so that is avoids any allocation.
+ * This function essentially computes:
+ * out_state = start_state + sum(scales[i] * derivs[i] for i = 1:N)
+ *
+ * out_state and start_state can be the same state.
+ *
+ * @tparam N Number of columns in the row
+ * @param out_state Where to save the new state
+ * @param start_state Starting state
+ * @param scales Derivative weights, one for each derivative state
+ * @param derivs State derivative values
+ */
+template<unsigned int N>
+constexpr void
+butcher_row(MoorDynState& out_state,
```

Comment:
I like this approach and the benefits we get. In-terms of work I doing, I can move my changes into this format quite easily. I have two additional states that have been added to lines, and the hackish way I've done it for now is to create another state vector that is a copy of the line state class. At the start of State,hpp:
```
class MoorDynState
{
  public:
	/// @brief Constructor
	MoorDynState() {}

	/// @brief Destructor
	~MoorDynState() {}

	/// The states of the lines
	std::vector<LineState> lines;

	/// The state derivatives of misc states: VIV, viscoelastic, blank (acc is dummy variable)
	std::vector<LineState> misc;
```

I realize this it not very efficient, but for timelines for the project that needed this work it was the fastest approach. When I open a PR with this we can discuss more.

Reply:
I agree that seems a little hacky, but if it works it works. 

If that's the primary change you're going to make to state stuff, then it should be pretty straightforward to integrate the state changes made here. 
When you make that PR I'll take a look and see if there's an easy opportunity for simplification/optimization.

Reply:
My first thought was to just change the vector sizes in the line state from vec3 for 5 (because these are technically line state variables) but everything in the state files is hard coded for vec3 or vec6, and it doesn't get around the fact that we only need to integrate these states once (unlike acceleration -> velocity -> position). I still need to develop tests for that work, and I am waiting on a preprint to link to explain the theory.